src/UpgradeRule/PHP/Visitor/RenameClassesVisitor.php

```diff
@@ -71,10 +146,42 @@ protected function handleStringUpdate(Scalar\String_ $stringNode)
                 return;
             }
 
-            // Substitute MyClass::class literal in place of string
             $baseName = $this->logUseStatement($replacement);
+
+            // Show warning if this replacement may be invalid
+            if (isset($this->renameWarnings[$baseName]) && $this->showPrompt) {
+
+                $before = $this->source->getNodeLine($stringNode, 'question');
+                $str = 'Attempting to rename: ' . PHP_EOL . $before . PHP_EOL
```

Comment:
Personally, I prefer `sprintf` for building these kind of strings, as I think the resulting code is easier to read. But that's more of a pet peeve.